tests/std/tests/P0660R10_jthread_and_cv_any/test.cpp

```diff
@@ -0,0 +1,244 @@
+    }
+
+    { // move assign
+        jthread worker_a{[] {}};
+        auto source_a = worker_a.get_stop_source();
+        jthread worker_b{[] {}};
+        auto id_b     = worker_b.get_id();
+        auto source_b = worker_b.get_stop_source();
+        worker_a      = move(worker_b);
+        assert(source_a.stop_requested());
+        assert(id_b == worker_a.get_id());
+        assert(!source_b.stop_requested());
+        assert(worker_a.get_stop_source() == source_b);
+        assert(!worker_b.joinable());
+    }
+
+    { // self move assign, currently specified to try to cancel and join
+        jthread worker{[] {}};
+        auto source = worker.get_stop_source();
+        worker      = move(worker);
+        assert(!worker.joinable());
+        assert(source.stop_requested());
+    }
+
+    { // swaps
+        jthread worker_a{[] {}};
+        auto id_a     = worker_a.get_id();
+        auto source_a = worker_a.get_stop_source();
+        auto token_a  = worker_a.get_stop_token();
+        jthread worker_b{[] {}};
+        auto id_b     = worker_b.get_id();
+        auto source_b = worker_b.get_stop_source();
+        auto token_b  = worker_b.get_stop_token();
+
+        assert(id_a != id_b);
+        assert(source_a != source_b);
+        assert(token_a != token_b);
+
+        worker_a.swap(worker_b);
+        assert(worker_a.get_id() == id_b);
+        assert(worker_a.get_stop_source() == source_b);
+        assert(worker_b.get_id() == id_a);
+        assert(worker_b.get_stop_source() == source_a);
+        swap(worker_a, worker_b);
+        assert(worker_a.get_id() == id_a);
+        assert(worker_a.get_stop_source() == source_a);
+        assert(worker_b.get_id() == id_b);
+        assert(worker_b.get_stop_source() == source_b);
+    }
+
+    { // join
+        jthread worker{[] {}};
+        auto source = worker.get_stop_source();
+        worker.join();
+        assert(!worker.joinable());
+        assert(worker.get_stop_source() == source);
+        assert(!source.stop_requested());
+        assert(source.stop_possible());
+    }
+
+    // TRANSITION, OS-11107628 "_Exit allows cleanup in other DLLs"
+    // detach() is intentionally not tested
+
+    // get_id, get_stop_source, get_stop_token tested above
+
+    assert(jthread::hardware_concurrency() == thread::hardware_concurrency());
+
+    { // first wait_until overload; without the cancellation this would deadlock
+        jthread worker([](stop_token token) {
+            mutex m;
+            condition_variable_any cv;
+            unique_lock lck{m};
+            assert(cv.wait(lck, move(token), [] { return false; }) == false);
+        });
+    }
+
+    { // ditto without the cancellation this would deadlock
+        jthread worker([](stop_token token) {
+            mutex m;
+            condition_variable_any cv;
+            unique_lock lck{m};
+            assert(cv.wait_until(lck, move(token), chrono::steady_clock::time_point::max(), [] { return false; })

```

Comment:
Suggestion: The max `time_point` and `duration` are verbose and repeatedly occurring, and they cause wrapping. Perhaps extract to `constexpr` variables?